core/src/main/java/org/apache/iceberg/view/ViewMetadata.java

```diff
@@ -157,6 +161,7 @@ private Builder() {
       this.history = Lists.newArrayList();
       this.properties = Maps.newHashMap();
       this.changes = Lists.newArrayList();
+      this.uuid = UUID.randomUUID().toString();
```

Comment:
Right, but this doesn't allow you to actually set the UUID because it always fails the check in `assignUUID`. If that always fails, then what's the point of adding the method?

I think this should start `null` in this case and allow assignment if it hasn't been initialized. Then initialize in `build`.

Reply:
should be fixed now